### PR TITLE
Fix tests for mac

### LINKED ISSUE #204: TestData should be platform independent
Right now, tests will fail to access `TestData` outside of Windows. There seem to be some path handling issues in `TestData.cs` and `PathUtils.cs`. The former hardcodes some `\` separated paths, and the latter seems to reimplement a bunch of things from `Path` and also do some trickery with forcing paths to be separated by `\`.

- Hardcoded path strings (like `src\UnitTests\TestData`) should be explicitly converted to the style of the platform running the tests.
- `PathUtils.cs` should be rewritten to not play around with the path separator, instead just being some helpers against the platform-independent path handling code from `Path`.

### LINKED ISSUE #205: PythonVersions cannot be constructed on macOS
The `GetCPythonVersion` calls in `PythonVersions.cs` throw `NullReferenceException`, starting from where tests access `Required_Python36X` (and other versions), i.e. when the static class gets constructed. This causes all of the tests which use this class (most of them) to fail, as they cannot access the `PythonVersion` instance needed to, say, construct a `Server`.

This prevents testing on macOS and likely also Linux. (`cd src; dotnet test`)

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Fix #204: TestData should be platform independent
- Fix #205: PythonVersions cannot be constructed on macOS

image

### CODE REVIEW

**Comment by jakebailey** [comment]
Partially related, but there are tests which assume that `sys.winver` exists, which then fail on non-Windows OSs. Probably good for another PR as we keep fixing tests.

File: src/Analysis/Engine/Test/UnixPythonInstallPathResolver.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            => architecture == InterpreterArchitecture.x86 ? null : _coreCache.TryGetValue(version, out var interpreterConfiguration) ? interpreterConfiguration : null;
+
+        public InterpreterConfiguration GetCondaPythonConfiguration(InterpreterArchitecture architecture, Version version)
+            => architecture == InterpreterArchitecture.x86 ? null : _condaCache.TryGetValue(version, out var interpreterConfiguration) ? interpreterConfiguration : null;
+
+        public InterpreterConfiguration GetIronPythonConfiguration(bool x64) => null;
+
+        private void GetConfigurationsFromKnownPaths() {
+            var homePath = Environment.GetEnvironmentVariable("HOME");
+            var knownFolders = new [] { "/usr/local/bin", "/usr/bin", "/bin", "/usr/sbin", "/sbin", "/usr/local/sbin" };
```
[the remark this answers is not in the document]
Comment (jakebailey) [reply src/Analysis/Engine/Test/UnixPythonInstallPathResolver.cs]:
```csharp
Environment.GetEnvironmentVariable("PATH")
    .Split(new[] { ':' }, StringSplitOptions.RemoveEmptyEntries)
```
Should probably do the job.

**Comment by AlexanderSher** [comment]
There are about 50 tests that have dependencies on windows-specific code. Maybe some of them should stay windows-specific and be simply ignored when executed on mac/unix.

File: src/Analysis/Engine/Test/UnixPythonInstallPathResolver.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakebailey) [reply src/Analysis/Engine/Test/UnixPythonInstallPathResolver.cs]: I just checked the bash manual to see what an empty item in `$PATH` does, and having an empty string item (`::`) in `$PATH` will actually mean the current directory, so your recent change of just using `Split(':')` is likely more correct than what I wrote above.

**Comment by jakebailey** [comment]
Yup, and I'm sure some of them are using those platform-specific items for purposes other than their platform-specific qualities (like the test that uses `kernel32.dll` because it's binary). Some of the `winver` tests may just use it for what it looks like when analyzed.

**Pull request merged**